client/my-sites/stats/site.jsx

```diff
@@ -135,49 +135,101 @@ Object.defineProperty( CHART_COMMENTS, 'label', {
-		const activeTab = getActiveTab( props.chartTab );
-		if ( activeTab !== state.activeTab ) {
-			return {
-				activeTab,
-				// TODO: remove this when we support hourly visitors.
-				activeLegend: props.period !== 'hour' ? activeTab.legendOptions || [] : [],
-			};
-		}
-		return null;
-	}
+	const [ activeTabState, setActiveTabState ] = useState( null );
+	const [ activeLegend, setActiveLegend ] = useState( null );
+	const { period } = props.period;
+
+	const queryDate = date.format( DATE_FORMAT );
+	const moduleStrings = statsStrings();
+
+	const isJetpack = useSelector( ( state ) => isJetpackSite( state, siteId ) );
+	const isOdysseyStats = config.isEnabled( 'is_running_in_jetpack_site' );
+	const isWPAdmin = config.isEnabled( 'is_odyssey' );
+
+	const slug = useSelector( getSelectedSiteSlug );
+	const upsellModalView = useSelector(
+		( state ) => config.isEnabled( 'stats/paid-wpcom-v2' ) && getUpsellModalView( state, siteId )
+	);
+
+	const {
+		supportsPlanUsage,
+		supportsUTMStats: supportsUTMStatsFeature,
+		supportsDevicesStats: supportsDevicesStatsFeature,
+		isOldJetpack,
+		supportUserFeedback,
+	} = useSelector( ( state ) => getEnvStatsFeatureSupportChecks( state, siteId ) );
+
+	const hasSiteLoadedFeatures = useSelector(
+		( state ) => isWPAdmin && hasLoadedSiteFeatures( state, siteId )
+	);
+
+	const shouldForceDefaultDateRange = useSelector( ( state ) =>
+		shouldGateStats( state, siteId, STATS_FEATURE_DATE_CONTROL_LAST_30_DAYS )
+	);
+
+	const shouldForceDefaultPeriod = useSelector( ( state ) =>
+		shouldGateStats( state, siteId, STATS_FEATURE_INTERVAL_DROPDOWN_WEEK )
+	);
+
+	const wpcomShowUpsell = useSelector(
+		( state ) =>
+			config.isEnabled( 'stats/paid-wpcom-v3' ) &&
+			shouldGateStats( state, siteId, STATS_FEATURE_PAGE_TRAFFIC )
+	);
+
+	const isAtomic = useSelector( ( state ) => isAtomicSite( state, siteId ) );
+	const isSitePrivate = useSelector( ( state ) => isPrivateSite( state, siteId ) );
+
+	const moduleToggles = useSelector( ( state ) => getModuleToggles( state, siteId, 'traffic' ) );
+	const momentSiteZone = useSelector( ( state ) => getMomentSiteZone( state, siteId ) );
+
+	const { supportedShortcutList } = useShortcuts();
 
-	getAvailableLegend() {
-		const { period } = this.props.period;
-		const activeTab = getActiveTab( this.props.chartTab );
+	const shouldShowUpsells = isOdysseyStats && ! isAtomic;
+	const supportsUTMStats = supportsUTMStatsFeature || isInternal;
+	const supportsDevicesStats = supportsDevicesStatsFeature || isInternal;
+
+	// Set up a custom range for the chart.
+	// Dependant on new date range picker controls.
+	let customChartRange = null;
+
+	const getAvailableLegend = () => {
+		const activeTab = getActiveTab( chartTab );
 		// TODO: remove this when we support hourly visitors.
 		return period !== 'hour' ? activeTab.legendOptions || [] : [];
-	}
+	};
 
-	navigationFromChartBar = ( periodStartDate, period ) => {
```

Comment:
Renamed `period` because there's a variable with the same name in the parent scope.